pynio/block.py

```diff
@@ -40,3 +40,11 @@ def name(self, value):
     @property
     def type(self):
         return self._type
+
+    def delete(self):
+        '''Delete self from instance and services'''
+        self._instance.blocks.remove(self._name)
+        for s in self._instance.services.values():
+            s._remove(self)
```

Comment:
Also, I prefer doing an instance.put -- this way all services + blocks are updated. This does have an issue if there are multiple users though.